app/uk/gov/hmrc/tradergoodsprofilesrouter/service/CreateProfileService.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.service
+
+import com.google.inject.Inject
+import play.api.Logging
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tradergoodsprofilesrouter.config.AppConfig
+import uk.gov.hmrc.tradergoodsprofilesrouter.connectors.{CreateProfileConnector, EisHttpErrorResponse}
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.request.CreateProfileRequest
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.request.eis.CreateProfileEisRequest
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.eis.CreateProfileResponse
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.errors.ErrorResponse
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ApplicationConstants.UnexpectedErrorCode
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CreateProfileService @Inject() (
+  connector: CreateProfileConnector,
+  uuidService: UuidService
+)(implicit
+  ec: ExecutionContext
+) extends Logging {
+
+  def createProfile(eori: String, request: CreateProfileRequest)(implicit
+    hc: HeaderCarrier
+  ): Future[Either[EisHttpErrorResponse, CreateProfileResponse]] = {
+    val eisRequest    =
+      CreateProfileEisRequest(
+        eori,
+        request.actorId,
+        request.ukimsNumber,
+        request.nirmsNumber,
+        padNiphlNumber(request.niphlNumber)
+      )
+    val correlationId = uuidService.uuid
+
+    connector
+      .createProfile(eisRequest, correlationId)
+      .map {
+        case Right(response) => Right(response)
+        case Left(response)  => Left(response)
```

Comment:
Would be good to have some logging here as well, so we can see what other responses EIS are sending